src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3084,12 +3084,16 @@ and genTypeDefn (TypeDef (ats, px, leadingKeyword, ao, tds, tcs, equalsRange, td
 
         typeName
         +> opt sepNone impCtor genMemberDefn
```

Comment:
If there is an `impCtor`, it will be printed twice here.
For example:
```fsharp
type NameStruct() =
    struct
    end
```
becomes
```fsharp
type NameStruct()() =
    struct
    end
```

but there is a bit more to this problem. I'll write it up in the summary.